status.go

```diff
@@ -86,6 +86,11 @@ func (statusList *StatusList) ByIndex(index int) (StatusEntry, error) {
 		return StatusEntry{}, ErrInvalid
 	}
 	ptr := C.git_status_byindex(statusList.ptr, C.size_t(index))
+	if ptr == nil {
+		runtime.LockOSThread()
+		defer runtime.UnlockOSThread()
+		return StatusEntry{}, MakeGitError(C.int(ErrNotFound))
```

Comment:
welp, unfortunately there's no consistency with what these functions return :(

`Blame.HunkByIndex`: https://github.com/libgit2/git2go/blob/462ebd83e0ccba9cd93c05ec12dc3d98064e3d5c/blame.go#L102
`Index.EntryByIndex`: https://github.com/libgit2/git2go/blob/462ebd83e0ccba9cd93c05ec12dc3d98064e3d5c/index.go#L420
`Tree.EntryByIndex`: Does nothing and may also need to be fixed (although since it returns a pointer, presumably it could return `nil` when that is fixed).

But in this case, since libgit2 does not set an error itself, using `MakeGitError()` will call [`giterr_last()`](https://github.com/libgit2/git2go/blob/462ebd83e0ccba9cd93c05ec12dc3d98064e3d5c/git.go#L281), which will most likely return `GIT_OK`, which will cause this error to have an empty message.

so I *think* the more correct solution here is to do the same thing as `Index.EntryByIndex`:

```suggestion
		return StatusEntry{}, fmt.Errorf("Index out of Bounds")
```